### ISSUE TITLE
System.Reflection.AmbiguousMatchException is thrown

### ISSUE THREAD

**Issue by joehmchan** [wojtpl2/ExtendedXmlSerializer#155]
System.Reflection.AmbiguousMatchException is thrown when 'Multiple custom attributes of the same type found.'

Here is the class definition.
```c#
   private object[] itemsField;

  [System.Xml.Serialization.XmlElementAttribute("Tenure", typeof(string), DataType = "integer", IsNullable = true)]
  [System.Xml.Serialization.XmlElementAttribute("Video360", typeof(URL), IsNullable = true)]
  [System.Xml.Serialization.XmlElementAttribute("VideoUrl", typeof(URL), IsNullable = true)]
  [System.Xml.Serialization.XmlElementAttribute("WebFeatures", typeof(TranslatableWebFeatures))]
  [System.Xml.Serialization.XmlChoiceIdentifierAttribute("ItemsElementName")]
  public object[] Items { 
    get { return this.itemsField; }
    set { this.itemsField = value; }
  }
```

It seems like multiple XmlElementAttribute decorations causing the issue.

**Comment by Mike-EEE** [comment]
Great, thanks for reporting this @joehmchan.  I am wrestling with Real Life/Work at the moment but hope to be back on this project in the next week or so to start tackling these issues. 😄

**Comment by joehmchan** [comment]
Yes, thanks for your help. I think the serializer creator should do special handling to XmlChoiceIdentifierAttribute

**Comment by Mike-EEE** [comment]
I am embarrassed to admit that I have not even heard of that attribute before. 😮 I will have to check it out.

**Comment by Mike-EEE** [comment]
OK @joehmchan I have checked in a fix for this.  This should unblock you if you are looking to serialize items without an exception. 😄   You can find this at our preview feed here:
https://www.myget.org/F/wojtpl2/api/v3/index.json

I did look into `XmlChoiceIdentifierAttribute` and boy this will take some further discussion.  Keep in mind that v2 (current version) was taking from v1, and I am sure it didn't support multiple attributes (@wojtpl2  can verify here).  Additionally the intended goal is to serialize objects of various, arbitrary types and ExtendedXmlSerializer does this with its own typing attribute system.

The one thing that it does not support is dynamic, runtime-based member naming.  There is a performance consideration here as it would have to look up the name for each and every member (or elements within a collection) whenever an object is emitted.

This would be a bit of work to implement, and I am wanting to throw it out there to see how important this is.  This is the first time I have seen this, so it would be good to know others' thoughts.

**Comment by wojtpl2** [comment]
V1 didn't support multiple attributes and System.Xml.Serialization.XmlChoiceIdentifierAttribute.

**Comment by Mike-EEE** [comment]
OK good to know.  That was my impression as well, @wojtpl2.

I've created a new item/discussion for this and we can continue this conversation there:
https://github.com/wojtpl2/ExtendedXmlSerializer/issues/159

Closing this item as it has been addressed/unblocked from throwing an exception.  Please feel free to reopen or open a new issue of there is any further issues that you find.
